src/main.ts

```diff
@@ -0,0 +1,258 @@
+import { hideBin } from 'yargs/helpers';
+import yargs from 'yargs/yargs';
+
+import { DEFAULT_TEMPLATE_REPOSITORY_NAME, USAGE_TEXT } from './constants';
+import type { MetaMaskRepository } from './establish-metamask-repository';
+import { establishMetaMaskRepository } from './establish-metamask-repository';
+import { lintProject } from './lint-project';
+import {
+  isPromiseFulfilledResult,
+  isPromiseRejectedResult,
+} from './misc-utils';
+import type { SimpleWriteStream } from './output-logger';
+import { OutputLogger } from './output-logger';
+import { reportProjectLintResult } from './report-project-lint-result';
+import { rules } from './rules';
+
+/**
+ * Data created from command-line input used to configure this tool.
+ */
+export type Inputs = {
+  templateRepositoryName: string;
+  projectReferences: string[];
+};
+
+/**
+ * The entrypoint for this tool. Designed to not access `process.argv`,
+ * `process.stdout`, or `process.stdout` directly so as to be more easily
+ * testable.
+ *
+ * @param args - The arguments to this function.
+ * @param args.argv - The name of this executable and its arguments (as obtained
+ * via `process.argv`).
+ * @param args.stdout - The standard out stream.
+ * @param args.stderr - The standard error stream.
+ * @param args.config - Extra configuration.
+ * @param args.config.cachedRepositoriesDirectoryPath - The directory where
+ * MetaMask repositories will be (or have been) cloned.
+ * @param args.config.defaultProjectNames - The set of MetaMask repositories
+ * that will be linted.
+ */
+export async function main({
+  argv,
+  stdout,
+  stderr,
+  config: { cachedRepositoriesDirectoryPath, defaultProjectNames },
+}: {
+  argv: string[];
+  stdout: SimpleWriteStream;
+  stderr: SimpleWriteStream;
+  config: {
+    cachedRepositoriesDirectoryPath: string;
+    defaultProjectNames: string[];
+  };
+}) {
+  const outputLogger = new OutputLogger({ stdout, stderr });
+  const workingDirectoryPath = process.cwd();
+
+  const inputs = await parseInputs({ argv, outputLogger, defaultProjectNames });
+  /* istanbul ignore next: At the moment, there is no real way that Yargs could fail */
```

Comment:
Is this branch still unreachable? Both here and in `parseInputs`, the branch where `null` is returned still has an ignore statement, but it looks like `parseCommandLineArguments` can return `null` (so it should be possible to test it here as well)

Reply:
Ah, you're right, this is now reachable. Not sure why I thought it wasn't before now 🤔  Removed here: [af59f53](https://github.com/MetaMask/module-lint/pull/33/commits/af59f5384afb4a21477964d071b770406c94842e)

Reply:
Thanks! I think you just missed it after turning on `strictOptions`